connection.go

```diff
@@ -188,6 +193,49 @@ func (c *Connection) Connect() error {
 	return nil
 }
 
+// ConnectCtx establishes the connection to the server using configured Addr
+func (c *Connection) ConnectCtx(ctx context.Context) error {
+	var conn net.Conn
+	var err error
+
+	if c.conn != nil {
+		return nil
+	}
```

Comment:
Usually Connect would be written in terms of ConnectCtx.

```go
func Connect() error {
   return ConnectCtx(context.Background())
}
func ConnectCtx(ctx context.Context) error {
   // so we only have one implementation of the logic
}
```

https://github.com/golang/go/blob/go1.22.3/src/database/sql/sql.go#L1657-L1676